contracts/WETH10.sol

```diff
@@ -1,102 +1,106 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
-pragma solidity 0.6.10;
-// Copyright (C) 2015, 2016, 2017 Dapphub / adapted by [] 2020
+pragma solidity 0.7.0;
+// Copyright (C) 2015, 2016, 2017 Dapphub / adapted by Ethereum Community 2020
 contract WETH10 {
-    string public name;
-    string public symbol;
-    uint8  public decimals;
-    bytes32 public DOMAIN_SEPARATOR;
-    bytes32 public PERMIT_TYPEHASH = keccak256("Permit(address src,address guy,uint wad,uint nonce,uint deadline)");
+    string public constant name = "Wrapped Ether";
+    string public constant symbol = "WETH";
+    uint8  public constant decimals = 18;
+    bytes32 public immutable DOMAIN_SEPARATOR;
+    bytes32 public immutable PERMIT_TYPEHASH = keccak256("Permit(address from,address spender,uint256 value,uint256 nonce,uint256 deadline)");
```

Comment:
This typehash does not follow the format specified in 2612.
It should be  `keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)")`

Reply:
Fixed this in #18